cpp/src/DataStorm/NodeSessionI.h

```diff
@@ -23,7 +22,12 @@ namespace DataStormI
         void destroy();
         void addSession(std::optional<DataStormContract::SessionPrx>);
 
-        std::optional<DataStormContract::NodePrx> getPublicNode() const { return _publicNode; }
+        DataStormContract::NodePrx getPublicNode() const
+        {
+            // always set after init
+            assert(_publicNode);
```

Comment:
Do we need this init function, can we move things into the constructor?

Reply:
Maybe we can, I will check in a follow up PR.